packages/connectivity/src/http-agent/http-agent.ts

```diff
@@ -114,24 +114,29 @@ function getTrustStoreOptions(destination: HttpDestination): {
 
 /**
  * @internal
- * The http agents (proxy and default) use node tls for the certificate handling. This method creates the options with the pfx and passphrase.
+ * The http agent uses node tls for the certificate handling. This method creates the options with the pfx and passphrase or key, cert and passphrase, depending on the format of the certificate.
  * https://nodejs.org/api/tls.html#tls_tls_createsecurecontext_options
- * @param destination - Destination object
+ * @param destination - Destination object.
  * @returns Options, which can be used later by tls.createSecureContext() e.g. pfx and passphrase or an empty object, if the protocol is not 'https:' or no client information are in the definition.
  */
-function getKeyStoreOptions(destination: Destination): {
-  pfx?: Buffer;
-  passphrase?: string;
-} {
+function getKeyStoreOptions(destination: Destination):
+  | {
+      pfx?: Buffer;
+      passphrase?: string;
+    }
+  | {
```

Comment:
Currently, every single parameter is optional, if one of the two options is chosen, isn't every single parameter required? Or can you have e.g. a pfx cert without passphrase or cert & key but no passphrase?

Reply:
Might even create a type for these two objects to make it a little more concise in the function API.